src/crimson/os/seastore/seastore_types.h

```diff
@@ -36,28 +36,148 @@ struct seastore_meta_t {
   }
 };
 
+// internal segment id type of segment_id_t, basically
+// this is a unsigned int with the top "device_id_len"
+// bits representing the id of the device on which the
+// segment resides
+using internal_segment_id_t = uint32_t;
+// type of device id
+using device_id_t = uint8_t;
+// order of device_id_t
+constexpr uint16_t device_id_len = 4;
+// maximum devices supported
+constexpr uint16_t max_devices = 1 << device_id_len;
+
+// segment ids without a device id encapsulated
+using device_segment_id_t = internal_segment_id_t;
```

Comment:
Because internal_segment_id_t is defined within segment_id_t, just declare this as

using device_segment_id_t = uint32_t.